### PR TITLE
CDAP-2088 Adding a TableSource for the ETL template. Also adding

### PR DESCRIPTION
**Pull request by albertshau**
a BatchReadableSource that the others extend so that there is a
catch-all for advanced users.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DT14

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/common/MockEmitter.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import co.cask.cdap.templates.etl.api.Emitter;
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+
+/**
+ * Mock emitter for unit tests
+ */
+public class MockEmitter<K, V> implements Emitter<K, V> {
+  private final Map<K, V> emitted = Maps.newHashMap();
```
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/common/MockEmitter.java]: Technically this should be List? I can emit multiple but identical <K, V> pairs. I guess it is fine for unit tests?

Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/common/MockEmitter.java]: good point, will change it

**Comment by gokulavasan** [comment]
One comment. Rest LGTM :+1:

**Pull request merged**